.github/workflows/ci.yml

```diff
@@ -126,16 +140,44 @@ jobs:
         working-directory: ./examples/non-trivial-type-on-stack
         run: cargo build
 
+  sanitizer:
+    name: Address Sanitizer
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      # Need nightly rust.
+      - uses: hecrj/setup-rust-action@v1
+        with:
+          rust-version: nightly
+          components: rust-src
+      - uses: Swatinem/rust-cache@v1
+      - name: Tests with asan
+        env:
+          RUSTFLAGS: -Zsanitizer=address -Cdebuginfo=0
+          RUSTDOCFLAGS: -Zsanitizer=address
+          ASAN_OPTIONS: "detect_stack_use_after_return=1:detect_leaks=0"
```

Comment:
Loving the look of this!

Reply:
> Loving the look of this!

Glad you like it. I don't think I have cracked it yet, but it's worth merging now anyway as it adds rustdoc, caching and cleans up the workflows a bit.

I'll keep working on the actual sanitizer part in another PR